<launch>
    <node name="load_pcl" pkg="pcl_tools" type="load_pcl" output="screen" >
        <param name="pcl2_output" type="string" value="/loaded_pcl2" />
        <param name="pcl_filepath" type="string" value="$(find pcl_tools)/doc/examples/load/test.pcd" />
        <param name="include_tf" type="bool" value="false" />
    </node>
    <arg name="search_radius" default="0.005" />
    <node name="pcl_to_mesh" pkg="pcl_tools" type="pcl_to_mesh" output="screen" >
        <param name="pcl2_input" type="string" value="/loaded_pcl2" />
        <param name="filepath" type="string" value="$(find pcl_tools)/doc/examples/mesh/test.obj" />

        <param name="threads" type="int" value="8" />
        <!-- 'GTP', 'Poisson', or 'MarchingCubes' -->
        <param name="algorithm" type="string" value="Poisson" />
            <!-- GPT variables -->
            <param name="GPT_search_radius" type="double" value="$(arg search_radius)" />
            <param name="GPT_mu" type="double" value="2.5" />
            <param name="GPT_nearest_neighbour" type="double" value="1000" />
            <param name="GPT_surface_angle" type="double" value="45" />
            <param name="GPT_min_angle" type="double" value="10" />
            <param name="GPT_max_angle" type="double" value="120" />
            <!-- Poisson variables -->
            <param name="Poisson_depth" type="double" value="14" />

        <param name="isPassthoughFilter" type="bool" value="false" />
        <param name="isMovingLeastSquares" type="bool" value="true" />
            <param name="MLS_search_radius" type="double" value="$(arg search_radius)" />
            <param name="MLS_polynomial_order" type="int" value="2" />
            <param name="MLS_upsampling_radius" type="double" value="0.005" />
            <param name="MLS_upsampling_step_size" type="double" value="0.003" />
        <param name="isNormalOMP" type="bool" value="true" />
            <param name="Normal_search_radius" type="double" value="$(arg search_radius)" />
    </node>
</launch>
